extensions/src/main/java/org/pentaho/platform/plugin/services/connections/PentahoSystemDriver.java

```diff
@@ -83,13 +84,23 @@ private String translate( String url ) {
   @Override
   public Connection connect( final String url, final Properties info ) throws SQLException {
     String translatedUrl = translate( url );
+    List<Exception> listOfExceptions = new ArrayList<>();
+
     for ( Driver driver : getAllDrivers() ) {
       if ( driver.acceptsURL( translatedUrl ) ) {
-        Connection conn = driver.connect( translatedUrl, info );
-        if ( conn != null ) {
-          return conn;
+        try {
+          Connection conn = driver.connect( translatedUrl, info );
+          if ( conn != null ) {
+            return conn;
+          }
+        } catch ( Exception e ) {
+          listOfExceptions.add( e );
         }
       }
+
+    }
+    if ( !listOfExceptions.isEmpty() ) {
+      throw new SQLException( listOfExceptions.get( 0 ) );
```

Comment:
Why not create a new custom Exception that extends SQLException that contains a list of all exceptions (listOfExceptions) instead of just return the first exception.

Reply:
Should we wrap around an unknown number of execptions? How many cases would we have of different exceptions. Do we need to complicate the way we handle these exceptions?